protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+# Catalog-owned tables
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/4381**
+
+This RFC proposes a new reader-writer table feature `catalogOwned` which changes the way Delta Lake
+accesses tables.
+
+Today’s Delta protocol relies entirely on the filesystem for read-time discovery as well as
+write-time commit atomicity. This feature request is to allow catalog-owned Delta tables whose
+discovery and commits go through the table's owning catalog instead of going directly to the
+filesystem (s3, abfs, etc). In particular, the catalog becomes the source of truth about whether a
+given commit attempt succeeded or not, instead of relying exclusively on filesystem PUT-if-absent
+primitives.
+
+Making the catalog the source of truth for commits to a table brings several important advantages:
+
+1. Allows the catalog to broker all commits to the tables it owns, and to reject filesystem-based
+   commits that would bypass the catalog. Otherwise, the catalog cannot reliably stay in sync with the
+   table state, nor can it reject invalid commits, because it doesn’t even know about writes until they
+   are already durable and visible to readers. For example, a catalog would want to block attempts to
+   drop the NOT NULL constraint on a column which is referenced by a FOREIGN KEY constraint in a
+   different table.
+
+2. Opens a clear path to transactions that could span multiple tables and/or involve non-table
+   catalog updates. Otherwise, the catalog cannot participate in commit at all, because
+   filesystem-based commits (i.e. using PUT-if-absent) do not admit any way to coordinate with other
+   entities.
+
+3. Allows the catalog to facilitate efficient writes of the table, e.g. by directly hosting the
+   content of small commits instead of forcing clients to write them to cloud storage first. Otherwise,
+   the catalog is not a source of truth, and at best it can only mirror stale copies of table state.
+
+4. Allows the catalog to facilitate efficient reads of the table. Examples include vending storage
```

Comment:
In theory the caching of such files doesn't need to be done by the catalog because they are immutable.

What is a great advantage of a catalog is that you can avoid having to do a LIST to get the latest committed version. That can save time, and it can also allow implementations of Delta on file systems where LIST is not ordered, such as S3 Express One Zone.